### PR TITLE
Feature/407209 classifier search query

### PR DESCRIPTION
**Pull request by PritiPiyushB**
### What one thing this PR does?

A sample one-liner about this task.

### Task details

- [354558 - Click to view X results on the layout](https://dev.azure.com/defragovuk/DEFRA-NCEA/_workitems/edit/354558/)
- [407209 - Implement UI changes for date guided search screen](https://dev.azure.com/defragovuk/DEFRA-NCEA/_workitems/edit/407209/)
- [354551 - View X results on layout](https://dev.azure.com/defragovuk/DEFRA-NCEA/_workitems/edit/354551)
-  [248899 - [Find] NCEA questionnaire search - geography data](https://dev.azure.com/defragovuk/DEFRA-NCEA/_workitems/edit/248899)
-  [408652 - Content updates](https://dev.azure.com/defragovuk/DEFRA-NCEA/_workitems/edit/408652/)

### How do these changes look like?
Screenshot 2024-07-24 at 07 24 02
Screenshot 2024-07-24 at 07 24 11
Screenshot 2024-07-24 at 07 24 25
Screenshot 2024-07-24 at 07 24 35

### Dev-tested in

1. Local environment

- [classifier category/link 1](https://dev-ncea-search.azure.defra.cloud//classifier-search?level=2&parent%5B%5D=lvl1-001)
- [classifier sub category/link 1](https://dev-ncea-search.azure.defra.cloud//classifier-search?level=3&parent%5B%5D=lv2-001)
- [date](https://dev-ncea-search.azure.defra.cloud//date-search?level=3&parent%5B%5D=lv3-001&cnt=382)
- [geography](https://dev-ncea-search.azure.defra.cloud//coordinate-search?level=3&parent%5B%5D=lv3-001&cnt=382)

2. Dev environment

- [classifier category/link 1](http://localhost:3000/classifier-search?level=2&parent%5B%5D=lvl1-001)
- [classifier sub category/link 1](http://localhost:3000/classifier-search?level=3&parent%5B%5D=lv2-001)
- [date](http://localhost:3000/date-search?level=3&parent%5B%5D=lv3-001&cnt=382)
- [geography](http://localhost:3000/coordinate-search?level=3&parent%5B%5D=lv3-001&cnt=382)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,6 +13,7 @@
     "build": "rimraf ./build && tsc && npm run copy-files",
     "copy-files": "ts-node copyNonTSFiles.ts",
     "json-server": "json-server ./src/services/mock-data.json --port 3300",
+    "classifier-json-server": "json-server --watch ./src/services/classifier-mock-data.json --port 3301",
```
Comment (karthigayayiniram) [inline package.json:16]: In an earlier PR, the micro service was integrated and we removed this.Why do we need to bring it back?

Comment (PritiPiyushB) [reply package.json]: Removed from package

**Review by karthigayayiniram — COMMENTED** [review]
please check on the updated design for the previous and skip buttons.
URL should always have the previous page parameters if any
local storage should be cleared when you go the home page(please refer the test env)

**Review by karthigayayiniram — COMMENTED** [review]
1. User clicks on 'Back' hyperlink on the journey, previous screen is not retained. Please refer test env.
2. User clicks on 'Restart questionnaire search' hyperlink on the search results screen, its not redirecting to the relevant screen.
3. Looks like the selected query params for level 3 is not passed to the payload. 
4. Payload query is not loading relevant values.
5. 
image

**Comment by karthigayayiniram** [comment]
Please address the following observations on your latest commit in order to ensure the positive flow is working as expected.

Back link is not working 
All disabled buttons ate enabled by default now
Filter options are displaying when there is no results (Query is incorrect as we discussed)

File: public/scripts/customScripts.js [hunk 2]
```diff
@@ -299,7 +469,11 @@ if (typeof Storage !== 'undefined') {
     previousQuestion();
     attachTodayDateEventListener();
     todayCheckboxStatus();
+    attachDateInputListeners();
     classifierBackLinkHandler();
+   // dateBackbreadcrumHandler();
```
Comment (karthigayayiniram) [inline public/scripts/customScripts.js:474]: remove it if not required

File: src/utils/queryBuilder.ts [hunk 3]
```diff
@@ -210,22 +216,36 @@ const _generateDateRangeQuery = (
...
 const generateSearchQuery = (searchBuilderPayload: ISearchBuilderPayload): estypes.SearchRequest => {
   const queryPayload: estypes.SearchRequest = _generateQuery(searchBuilderPayload);
   const { searchFieldsObject, docId = '' } = searchBuilderPayload;
   const { filters } = (searchFieldsObject as ISearchPayload) ?? {};
+  const { level, parent } = (searchFieldsObject?.fields.classify as ISearchPayload) ?? {};
+  const levelMap = {
+    1: 'OrgNceaClassifiers.code.keyword',
+    2: 'OrgNceaClassifiers.classifiers.code.keyword',
+    3: 'OrgNceaClassifiers.classifiers.classifiers.code.keyword',
+  };
```
Comment (karthigayayiniram) [inline src/utils/queryBuilder.ts:234]: move this const levelMap to a common place and use it. code duplicate

File: src/utils/queryBuilder.ts [hunk 4]
```diff
@@ -240,6 +260,13 @@ const _generateStudyPeriodFilterQuery = (searchBuilderPayload: ISearchBuilderPay
   const queryPayload: estypes.SearchRequest = _generateQuery(searchBuilderPayload);
   const { searchFieldsObject, docId = '' } = searchBuilderPayload;
   const { fields, filters } = searchFieldsObject as ISearchPayload;
+  const { level, parent } = (searchFieldsObject?.fields.classify as ISearchPayload) ?? {};
+  const levelMap = {
+    1: 'OrgNceaClassifiers.code.keyword',
+    2: 'OrgNceaClassifiers.classifiers.code.keyword',
+    3: 'OrgNceaClassifiers.classifiers.classifiers.code.keyword',
+  };
```
Comment (karthigayayiniram) [inline src/utils/queryBuilder.ts:268]: duplicate code.

**Review by karthigayayiniram — APPROVED** [review]
few review comments can be addressed in the upcoming PR.

File: src/utils/queryBuilder.ts (hunk 3 above)
Comment (PritiPiyushB) [reply src/utils/queryBuilder.ts:234]: Will address this in next pr

File: src/utils/queryBuilder.ts (hunk 4 above)
Comment (PritiPiyushB) [reply src/utils/queryBuilder.ts:268]: Will address this in next pr

**Review by ganesh-rajamahanthy — APPROVED** [review]

**Pull request merged**